### PR TITLE
Process data in models

### PR DESCRIPTION
**Pull request by dennis-every**
In this pull request we completed the following requirements:

### Create models
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Remember that author_id column in posts table should be the foreign_key for the users table.
- Use models to insert data using the rails console

### Create custom methods
Your models should include some additional methods.
#### Users
- A method that returns the 3 most recent posts for a given user.
#### Posts
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.

#### Comments
- A method that updates the comments counter for a post.

#### Likes
- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 class Like < ApplicationRecord
-  belongs_to :post
+  belongs_to :post, counter_cache: :likes_counter
+  belongs_to :author, class_name: 'User'
 end
```
Comment (julie-ify) [inline app/models/like.rb]: - [ ] There should be a method for incrementing the likes count of a post after a like has been added.

- [ ] **[OPTIONAL]** I would also suggest you add an ActiveRecord callback, like `after_save`, for calling the method whenever a like is created.

  ```
  class Like < ApplicationRecord
    belongs_to :post, counter_cache: :likes_counter
    belongs_to :author, class_name: 'User'
    after_save :update_post_likes_counter
 
    private
    
    def update_post_likes_counter
      post.increment!(:likes_counter)
    end
  end
  ```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 class Comment < ApplicationRecord
-  belongs_to :post
+  belongs_to :post, counter_cache: :comments_counter
+  belongs_to :author, class_name: 'User'
 end
```
Comment (julie-ify) [inline app/models/comment.rb]: - [ ] There should be a method for incrementing the comments count of post after a comment has been added.

- [ ] **[OPTIONAL]** I would also suggest you add an ActiveRecord callback, like `after_save`, for calling the method whenever a comment is created.

  ```
  class Comment < ApplicationRecord
    belongs_to :post
    belongs_to :author, class_name: 'User'
    after_save :update_post_comments_counter
 
    private
    
    def update_post_comments_counter
      post.increment!(:comments_counter)
    end
  end
  ```

File: app/models/post.rb [hunk 3]
```diff
@@ -1,5 +1,9 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  belongs_to :author, class_name: 'User', counter_cache: :posts_counter
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  def last_5_comments
+    comments.order(created_at: :desc).limit(5)
+  end
 end
```
Comment (julie-ify) [inline app/models/post.rb]: - [ ] There should be a method for incrementing the post count of a user after a post has been added.

- [ ] **[OPTIONAL]** I would also suggest you add an ActiveRecord callback, like `after_save`, for calling the method whenever a post is created.

  ```
  class Post < ApplicationRecord
    belongs_to :author, class_name: 'User', counter_cache: :posts_counter
    has_many :comments, dependent: :destroy
    has_many :likes, dependent: :destroy
    after_save :update_posts_counter
  
    def last_5_comments
      comments.order(created_at: :desc).limit(5)
    end
 
    private
    
    def update_post_counter
      author.increment!(:posts_counter)
    end
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @dennis-every,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @dennis-every , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**